server/app/controllers/admin/AdminApiKeysController.java

```diff
@@ -0,0 +1,77 @@
+package controllers.admin;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.Authorizers;
+import auth.CiviFormProfile;
+import auth.ProfileUtils;
+import controllers.CiviFormController;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.pac4j.play.java.Secure;
+import play.data.DynamicForm;
+import play.data.FormFactory;
+import play.mvc.Http;
+import play.mvc.Result;
+import services.apikey.ApiKeyCreationResult;
+import services.apikey.ApiKeyService;
+import services.program.ProgramService;
+import views.admin.apikeys.ApiKeyNewOneView;
+import views.admin.programs.ApiKeyCredentialsView;
+
+/** Controller for admins managing ApiKeys. */
+public class AdminApiKeysController extends CiviFormController {
+
+  private final ApiKeyService apiKeyService;
+  private final ApiKeyNewOneView newOneView;
+  private final ApiKeyCredentialsView apiKeyCredentialsView;
+  private final ProgramService programService;
+  private final FormFactory formFactory;
+  private final ProfileUtils profileUtils;
+
+  @Inject
+  public AdminApiKeysController(
+      ApiKeyService apiKeyService,
+      ApiKeyNewOneView newOneView,
+      ApiKeyCredentialsView apiKeyCredentialsView,
+      ProgramService programService,
+      FormFactory formFactory,
+      ProfileUtils profileUtils) {
+    this.apiKeyService = checkNotNull(apiKeyService);
+    this.newOneView = checkNotNull(newOneView);
+    this.apiKeyCredentialsView = checkNotNull(apiKeyCredentialsView);
+    this.programService = checkNotNull(programService);
+    this.formFactory = checkNotNull(formFactory);
+    this.profileUtils = checkNotNull(profileUtils);
+  }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result index() {
+    return ok();
+  }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result newOne(Http.Request request) {
+    return ok(newOneView.render(request, programService.getActiveProgramNames()));
+  }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result create(Http.Request request) {
+    Optional<CiviFormProfile> profile = profileUtils.currentUserProfile(request);
+
+    if (!profile.isPresent()) {
+      throw new RuntimeException("Unable to resolve profile.");
+    }
+
+    DynamicForm form = formFactory.form().bindFromRequest(request);
+    ApiKeyCreationResult result = apiKeyService.createApiKey(form, profile.get());
+
+    if (result.isSuccessful()) {
```

Comment:
Not-blocking: Complete style nit. Consider putting the error handling in the if-block and putting the "happy" case as the last statement. This is mostly a style preference that reading through the code ends in the case we expect. A perhaps weaker argument is that there's another indented block in this code which throws an exception, so it's perhaps slightly less readable to have the next indented block actually return the success case. My preference is pretty weak here, so feel free to discard if you'd like.

Reply:
hahah that was actually its original structure, @shanemc-goog requested a change to avoid having `if (!`. i don't have a strong opinion either.

Reply:
hah lol yeah.  Happy either way, I can see brett's case too.